wordpress-mobile/WordPress-Android
# Homepage settings in page list

Related FluxC PR - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1583

This PR introduces functionality to set a homepage or a posts page from the pages list. The two methods have been added to the popup menu on each page item in the `Published` list. 

To test:

1. Use a WPCom site for this test
2. Set up your site in Calypso/Design/Customize/Homepage settings to have the "Static page" shown on the front page
3. Make sure you have multiple pages created for the site
4. Log in with this site in the app
5. Go to Pages
6. Click on the overflow menu
7. Check that you see both the "Set as Homepage" and "Set as Posts Page" options
8. Click on "Set as Homepage" 
9. Click on the overflow menu on the same page
10. Check that the "Set as Homepage" option is no longer available

To test:
Use steps 1-7 from the previous test
8. Click on "Set as Posts Page"
9. Click on the overflow menu on the same page
10. Check that the "Set as Posts page" option is no longer available

To test:
Use steps 1-7 from the previous test
8. Click on "Set as Posts Page"
9. Click on the overflow menu on the same page
10. Check that the "Set as Posts page" option is no longer available
11. Click on "Set as Homepage"
12. Click on the overflow menu on the same page
13. See that the "Set as Homepage" is no longer available but the "Set as Posts Page" is now available

To test: 
- Repeat all the previous steps with a self-hosted site connected through Jetpack

To test:
1. Use a Self-hosted site without Jetpack for this test
Use steps 2 - 6
7. Check that the options "Set as Posts Page" and "Set as Homepage" are missing from the overflow menu

To test:
1. Use a WPCom site for this test
2. Set up your site in Calypso/Design/Customize/Homepage settings to have the "Newest posts" shown on the front page
Use steps 3 - 6
7. Check that the options "Set as Posts Page" and "Set as Homepage" are missing from the overflow menu

PR submission checklist:
